### DISCUSSION TITLE
The Plan

### DISCUSSION THREAD

**Discussion by soupday** [soupday/cc3_blender_tools#4] [General]
## Overview
Ultimately I am trying to create a CC3/iClone <> Blender > Unity+others pipeline:

![plan_01](https://user-images.githubusercontent.com/79094830/111026421-0edacd00-83e2-11eb-89f3-a217f8822be0.jpg)

The reason for this is as follows:
 - To be able to bring all of Blenders rendering power to bear on characters made in CC3 and/or scenes from iClone, e.g. to create high quality character artworks.
 - Similarly, with Unity: To bring CC3 characters to Unity in the highest quality possible without sacrificing performance.
 - I'm not happy with the results of the Reallusion made Unity auto-setup, it ignores or many of the subtle details that CC3 is capable of: blend maps, cavity occlusion, detail overlays, etc. The resulting characters in Unity look flat and generic. I can get much better results processing the textures for Unity by hand, but it is an _extraordinarily_ slow process.
 - It's rare that the exports from CC3/iClone to other platforms don't need at least some kind of alteration to fix mesh problems or vertex weights. This is unavoidable, different platforms do things... differently.
 - Once you take a character out of the CC3/iClone ecosystem everything just seems to slow down to an absolute crawl. I aim to speed that up, even if it's just for my own use.

I envisage achieving this with 3 interconnected add-ons:
 - One add-on to import and set-up the materials and physics of a character from CC3 or iClone.
 - Another add-on to help with rigging, animation and rendering.
 - And finally an exporter add-on to process and bake textures, materials and meshes for export to other platforms, specifically Unity, but other platforms are possible.

**Comment by soupday** [comment]
Something like this, anyway. It remains to be seen how well it works in practice.

**Reply by greggpatton** [reply]
I like what you've laid out. 

I have been dabbling in Blender, Unity, CC3, iClone, ZBrush for years. I'm currently trying to figure out a workflow to illustrate a children's story I've written. Blender, CC3, a little bit of ZBrush and, the use of image filters are where it's at right now. I was going to use iClone instead of Blender to create my base images until I ran across your project. I much prefer Blender but getting CC3 models to look good in Blender was making it a little too daunting for me. Happily, that is no longer an issue!

**Comment by edpark** [comment]
Hi there, I just wanted to say _thank you_ for creating this add-on it's awesome! Is there some way I can make a donation? Buy you a virtual beer? 😃

**Comment by mastajappa** [comment]
Thanks so much for your effort! Keep up the great work.

**Comment by VirtualGirl666** [comment]
I have created an account just to thank you for your amazing work, I've been following your progress for a while now and I'm truly thankfull for what you've done! I have a question, tho: Are you planning to make it compatible with blender Octane in the futur ?

Cordially!

**Comment by utanipishtim** [comment]
Your work has been nothing short of amazing! Thank you SO much for all the time and effort you have dedicated to this.  The results speak for themselves - just FANTASTIC!
